File: src/main/java/cookiedragon/luchadora/module/impl/player/MultiTaskModule.java
package cookiedragon.luchadora.module.impl.player;

import cookiedragon.luchadora.event.api.Subscriber;
import cookiedragon.luchadora.event.entity.AllowInteractEvent;
import cookiedragon.luchadora.module.AbstractModule;
import cookiedragon.luchadora.module.Category;

/**
 * @author cookiedragon234 31/Dec/2019
 */
@AbstractModule.Deceleration(name = "MultiTask", description = "Peform multiple actions at once", category = Category.PLAYER)
public class MultiTaskModule extends AbstractModule
{
	@Subscriber
	private void allowInteract(AllowInteractEvent event)
	{
		event.isUsingItem = false;
	}
	
	@Override
	protected void onEnabled()
	{
	
	}
	
	@Override
	protected void onDisabled()
	{
	
	}
}

File: src/main/kotlin/cookiedragon/luchadora/module/impl/player/MultiTaskModule.kt
package cookiedragon.luchadora.module.impl.player

import cookiedragon.luchadora.event.api.Subscriber
import cookiedragon.luchadora.event.entity.AllowInteractEvent
import cookiedragon.luchadora.module.AbstractModule
import cookiedragon.luchadora.module.Category

/**
 * @author cookiedragon234 11/Jan/2020
 */
class MultiTaskModule : AbstractModule("MultiTask", "Perform multiple actions at once", Category.PLAYER) {
	@Subscriber
	private fun allowInteract(event: AllowInteractEvent) {
		event.isUsingItem = false
	}
}

